src/integrate/orchestrator.rs

```diff
@@ -0,0 +1,1204 @@
+rpcport={}
+"#,
+            devnet_config.bitcoin_node_username,
+            devnet_config.bitcoin_node_password,
+            devnet_config.bitcoin_node_p2p_port,
+            devnet_config.bitcoin_node_rpc_port,
+            devnet_config.bitcoin_node_rpc_port,
+        );
+        let mut bitcoind_conf_path = PathBuf::from(&devnet_config.working_dir);
+        bitcoind_conf_path.push("conf/bitcoin.conf");
+        let mut file = File::create(bitcoind_conf_path).expect("Unable to create bitcoind.conf");
+        file.write_all(bitcoind_conf.as_bytes())
+            .expect("Unable to write bitcoind.conf");
+
+        let bitcoin_controller_conf = format!(
+            r#"
+[network]
+rpc_bind = "0.0.0.0:{}"
+block_time = {}
+miner_address = "{}"
+bitcoind_rpc_host = "0.0.0.0:{}"
+bitcoind_rpc_user = "{}"
+bitcoind_rpc_pass = "{}"
+whitelisted_rpc_calls = [
+    "listunspent",
+    "listwallets",
+    "createwallet",
+    "importaddress",
+    "sendrawtransaction",
+    "getrawtransaction",
+    "scantxoutset",
+    "getrawmempool",
+    "getblockhash",
+]
+# Expedite the key registration / genesis block
+[[blocks]]
+count = 1
+block_time = 30000
+ignore_txs = false
+
+# Expedite the key registration / genesis block
+[[blocks]]
+count = 2
+block_time = 20000
+ignore_txs = false
+
+# Give more time to the first blocks
+[[blocks]]
+count = 3
+block_time = 45000
+ignore_txs = false
+"#,
+            devnet_config.bitcoin_controller_port,
+            devnet_config.bitcoin_controller_block_time,
+            devnet_config.miner_btc_address,
+            devnet_config.bitcoin_node_rpc_port,
+            devnet_config.bitcoin_node_username,
+            devnet_config.bitcoin_node_password,
+        );
+        let mut bitcoin_controller_conf_path = PathBuf::from(&devnet_config.working_dir);
+        bitcoin_controller_conf_path.push("conf/puppet-chain.toml");
+
+        let mut file = File::create(bitcoin_controller_conf_path)
+            .expect("Unable to create bitcoin_controller.toml");
+        file.write_all(bitcoin_controller_conf.as_bytes())
+            .expect("Unable to create bitcoin_controller.toml");
+
+        let mut exposed_ports = HashMap::new();
+        exposed_ports.insert(
+            format!("{}/tcp", devnet_config.bitcoin_controller_port),
+            HashMap::new(),
+        );
+        exposed_ports.insert(
+            format!("{}/tcp", devnet_config.bitcoin_node_rpc_port),
+            HashMap::new(),
+        );
+        exposed_ports.insert(
+            format!("{}/tcp", devnet_config.bitcoin_node_p2p_port),
+            HashMap::new(),
+        );
+
+        let mut labels = HashMap::new();
+        labels.insert("project".to_string(), self.network_name.to_string()); 
+
+        let config = Config {
+            labels: Some(labels),
+            image: Some(devnet_config.bitcoin_node_image_url.clone()),
+            domainname: Some(self.network_name.to_string()),
+            tty: Some(true),
```

Comment:
Curious, what purpose is the TTY serving in `devnet`? Is it necessary for some operations?

Reply:
Addressed with latest